include/deal.II/base/work_stream.h

```diff
@@ -1089,6 +1090,73 @@ namespace WorkStream
   }
 
 
+
+  /**
+   * Same as the function above, but for iterator ranges. A class that fulfills
+   * the requirements of an iterator range defines the functions
+   * `IteratorRange::begin()` and `IteratorRange::end()`, both of which return
+   * iterators to elements that form the bounds of the range.
+   */
+  template <typename Worker,
+            typename Copier,
+            typename IteratorRange,
+            typename ScratchData,
+            typename CopyData>
+  void
+  run(const IteratorRange &iterator_range,
```

Comment:
why pass by const ref instead of by value? That would eliminate the const_cast below...

Reply:
Good tip. Thanks a lot! That's a lot better :-)